### PR TITLE
feat: Redesign for the integrations pages

### PR DESCRIPTION
**Pull request by ivasilov**

### CODE REVIEW

**Review by alaister — COMMENTED** [review]
some icons in light more are still light:
Screenshot 2024-11-22 at 13 11 28
Screenshot 2024-11-22 at 13 11 46



missing empty state for postgres extensions:
Screenshot 2024-11-22 at 13 10 24

File: apps/studio/components/interfaces/Integrations/NewQueues/QueuesRows.tsx [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+  const { project: selectedProject } = useProjectContext()
+
+  const { data: metrics, isLoading } = useQueuesMetricsQuery(
+    {
+      queueName: queue.queue_name,
+      projectRef: selectedProject?.ref,
+      connectionString: selectedProject?.connectionString,
+    },
+    {
+      staleTime: 30 * 1000, // 60 seconds, talk with Oli whether this is ok to call every minute
```
Comment (alaister) [inline apps/studio/components/interfaces/Integrations/NewQueues/QueuesRows.tsx:28]:
```suggestion
      staleTime: 30 * 1000, // 30 seconds, talk with Oli whether this is ok to call every minute
```

File: apps/studio/components/interfaces/Integrations/Wrappers/WrappersTab.tsx [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+  const { data, isLoading } = useFDWsQuery({
+    projectRef: project?.ref,
+    connectionString: project?.connectionString,
+  })
+
+  const wrappers = data?.result || []
+
+  if (isLoading) {
+    return <div>Loading</div>
```
Comment (alaister) [inline apps/studio/components/interfaces/Integrations/Wrappers/WrappersTab.tsx:32]: Does this loading state get shown? Should we make it a bit nicer?

Comment (joshenlim) [reply apps/studio/components/interfaces/Integrations/Wrappers/WrappersTab.tsx:32]: actually good point - this never gets called cause the loading state is handled by the parent already, will remove

**Review by alaister — COMMENTED** [review]
Tested:
- viewing integrations home page
- filtering integrations (search)
- viewing integration page
- misc
  - **the installed integrations is missing a loading skeleton**
- cron
  - creating a new job
  - filtering jobs
  - activating/deactivating
  - editing
  - view previous runs
    - **browser back button doesn't work properly after clicking this**
  - deleting a cron job
- GraphiQL
- Vault
  - **the default search under keys is _vault_**
  - create/delete key
  - crud for secrets
    - **on editing a secret, the value is missing even though I can see it on the list view**
  - sorting keys/secrets
- webhooks
  - filtering
  - crud
- wrappers
  - crud on both wrappers and tables
  - view encryption key in vault
    - **small one but clicking this makes the whole ui loading trigger**
    - **I did also notice one time on edit it didn't fetch my secret. can't replicate now though**
  - **visual issue once you have a few tables**:
Screenshot 2024-11-25 at 15 12 12

File: apps/studio/components/interfaces/Integrations/NewQueues/QueuesRows.tsx [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+  return (
+    <Table.tr
+      key={queue.queue_name}
+      onClick={() => {
+        router.push(
+          `/project/${selectedProject?.ref}/integrations/queues/queues/${queue.queue_name}`
+        )
+      }}
+      className="hover:"
```
[the remark this answers is not in the document]
Comment (joshenlim) [reply apps/studio/components/interfaces/Integrations/NewQueues/QueuesRows.tsx:42]: hmm will remove these!

[the remark this answers is not in the document]
Comment (joshenlim) [reply apps/studio/components/interfaces/Integrations/NewQueues/QueuesRows.tsx:42]: actually this file also looks duped - will remove

[the remark this answers is not in the document]
Comment (joshenlim) [reply apps/studio/components/interfaces/Integrations/NewQueues/QueuesRows.tsx:42]: actually i'll just clean up the class names - will let Ivan do the clean up of the files in a separate PR

**Comment by joshenlim** [comment]
> browser back button doesn't work properly after clicking this

This is a known issue for all links to the SQL editor that has the content param if im not wrong - will address separately (but either way i think we should have a custom UI for previous runs anyways, not just link to SQL editor)

**Review by staaldraad — APPROVED** [review]
approved `apps/studio/next.config.js` changes

**Review by alaister — APPROVED** [review]

**Pull request merged**